library/src/main/java/com/doctoror/rxcursorloader/RxCursorLoaderFlowableFactory.java

```diff
@@ -91,20 +90,17 @@ public void run() {
                 @NonNull final Scheduler scheduler) {
             mContentResolver = resolver;
             mQuery = query;
-            this.mScheduler = scheduler;
+            mScheduler = scheduler;
         }
 
         @Override
         public void subscribe(final FlowableEmitter<Cursor> emitter) {
-            final HandlerThread handlerThread = new HandlerThread(TAG.concat(".HandlerThread"));
-            handlerThread.start();
             synchronized (mLock) {
-                mHandler = new Handler(handlerThread.getLooper());
                 mEmitter = emitter;
                 mContentResolver.registerContentObserver(mQuery.contentUri, true,
                         getResolverObserver());
+                reload();
```

Comment:
Double locking detected. Should not move this `reload` invocation inside synchronized block.

Reply:
There is no double lock when using `synchronized`. [It is reentrant](http://tutorials.jenkov.com/java-concurrency/locks.html#reentrance)